nemo/core/connectors/save_restore_connector.py

```diff
@@ -37,7 +39,7 @@ def __init__(self) -> None:
         self._model_weights_ckpt = "model_weights.ckpt"
         self._model_extracted_dir = None
 
-    def save_to(self, model, save_path: str):
+    def save_to(self, model: nemo_classes.ModelPT, save_path: str):
```

Comment:
Just use string here, do not use the actual class just for typing

Reply:
I think we should avoid quotes, and use [PEP563](https://peps.python.org/pep-0563/) as a more stable approach, since quotes are intended to use with forward declarations by original [PEP484](https://peps.python.org/pep-0484/#forward-references), and there is no forward declaration here. 

But there is still a problem with partial initialization, so I used `from __future__ import annotations`. This is fully compatible with Python3.7+, widely recommended and adopted.

E.g. see in [mypy](https://mypy.readthedocs.io/en/stable/runtime_troubles.html#future-annotations-import-pep-563), [pydantic](https://docs.pydantic.dev/usage/postponed_annotations/) and so on.

You can't see it in old code (due to Python3.7+ limitations), but in new code it is also widely use it, e.g.:
- [Lightning (app)](https://github.com/Lightning-AI/lightning/blob/10a4b245291c986984885ed50aba830318305674/src/lightning_app/frontend/panel/app_state_watcher.py)
- [TorchVision](https://github.com/pytorch/vision/blob/1353df9e45aae6739c23db4a053c3d9a9fd17b3d/torchvision/prototype/datapoints/_datapoint.py)

To avoid long discussions here, **I added quotes**. 

But I will appreciate if you look at alternative approach and discuss it (maybe separately).

Reply:
Please use string name here. Pep is not an enforcement it is a suggestion. It does not make sense to mess around internals of Nemo just for an annotation.